backend/OpenAPI.yaml

```diff
@@ -248,22 +242,293 @@ paths:
+                    pictures: ["https://example.com/image1.jpg"]
+                    category: "Electronics"
+                    condition: "Used"
+                    date_posted: "2025-02-22T10:00:00Z"
+                    campus: "UTM"
+                    total: 1
+        "400":
+          description: Bad Request - Missing or invalid user_id
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        "404":
+          description: Saved items not found for the specified user
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        "500":
+          description: Internal Server Error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"  
+    delete:
+      summary: Delete a saved item
+      description: Remove a listing from a user's saved items collection.
+      parameters:
+        - name: saved_item_id
+          in: path
+          required: true
+          description: The unique identifier of the saved item to delete.
+          schema:
+            type: string
+        - name: user_id
+          in: query
+          required: true
+          description: The unique identifier of the user who owns the saved item.
+          schema:
+            type: string
+      responses:
+        "200":
+          description: Saved item successfully deleted.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                example:
+                  message: "Saved item deleted successfully."
+        "400":
+          description: Bad Request - Missing or invalid saved_item_id or user_id.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        "404":
+          description: Saved item not found.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        "500":
+          description: Internal Server Error.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+
+  /messages:
+    post:
+      summary: Send a message
+      description: Create and send a new message from one user to another.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/MessagesPostRequest"
+      responses:
+        "201":
+          description: Message sent successfully.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/MessagesPostResponse"
+        "400":
```

Comment:
Use the standard 422 response for this. Bad POST bodies are automatically handled as a 422 error with FastAPI